src/server/common.cc

```diff
@@ -24,11 +25,31 @@ extern "C" {
 #include "server/server_state.h"
 #include "server/transaction.h"
 
+ABSL_FLAG(bool, lock_on_hashtags, false,
+          "When true, locks are done in the {hashtag} level instead of key level. "
+          "Only use this with --cluster_mode=emulated|yes.");
+
 namespace dfly {
 
 using namespace std;
 using namespace util;
 
+bool KeyLockArgs::IsLockHashTagEnabled() {
+  thread_local bool is_enabled_flag_cache = []() {
+    return absl::GetFlag(FLAGS_lock_on_hashtags);
+  }();
+
+  return is_enabled_flag_cache;
+}
+
+string_view KeyLockArgs::GetLockKey(string_view key) {
+  if (IsLockHashTagEnabled()) {
+    return ClusterConfig::KeyTag(key);
+  } else {
+    return key;
+  }
```

Comment:
```suggestion
    return key;
```

Reply:
small nit but even clang tidy has a flag for this https://clang.llvm.org/extra/clang-tidy/checks/readability/else-after-return.html

Reply:
I actually think it's less readable, but given the smart folks of clang disagree I'll obey :)